pages/plugins/writing.md.erb

```diff
@@ -46,10 +49,10 @@ configuration:
 
 The `configuration` property defines the validation rules for the plugin configuration using the [JSON Schema](https://json-schema.org) format. The plugin in this tutorial has a single `pattern` property, of type `string`.
 
-Configuration properties are available to the hook script as environment variables with the naming pattern `BUILDKITE_PLUGIN_<PLUGIN_NAME>_<CONFIGURATION_PROPERTY>` where `<PLUGIN_NAME>` is the GitHub repository slug, not the name defined in `plugin.yml`. In this case, the configured value of `pattern` will be available as `BUILDKITE_PLUGIN_FILE_COUNTER_PATTERN`.
+Configuration properties are available to the hook script as environment variables with the naming pattern `BUILDKITE_PLUGIN_<PLUGIN_NAME>_<CONFIGURATION_PROPERTY>` where `<PLUGIN_NAME>` is not the name defined in `plugin.yml` but an uppercase version of the plugin's ID. In this case, the configured value of `pattern` will be available as `BUILDKITE_PLUGIN_FILE_COUNTER_PATTERN`.
```

Comment:
Is the _plugin ID_ defined or referenced anywhere else? Does it mean a lowercase version of the plugin name, with any whitespace replaced with dashes?

Reply:
That is a good point... I don't think that there is a full documentation of how the ID is created. Environment variables are [created by the agent](https://github.com/buildkite/agent/blob/ec8675e062f4c9e83e772a827305b66331f1ba7a/agent/plugin/plugin.go#L249). Following the code, it would appear that - at this time at least - the logic is as follows:

* take [the plugin name](https://github.com/buildkite/agent/blob/ec8675e062f4c9e83e772a827305b66331f1ba7a/agent/plugin/plugin.go#L139):
  - from the plugin's definition, take whatever is after the last slash (`/`)
  - make it all lowercase
  - replace every whitespace in it with a single space
  - replace every character that is not a number or letter with a dash `-`
  - remove the `-buildkite-plugin(-git)` suffix
* convert it to a [variable name](https://github.com/buildkite/agent/blob/ec8675e062f4c9e83e772a827305b66331f1ba7a/agent/plugin/plugin.go#L207):
  - make it all uppercase
  - replace every whitespace in it with a single space
  - dashes and spaces are replaced with underscores
  - if there are contiguous underscores, they are replaced with a single one

So, basically, it is mostly what you said... except it is not the plugin name, but how it is referenced in the pipeline itself.

I used `plugin id` because that is how the linter defines it, but I don't think that it makes sense to use it.

How about the following?

```suggestion
Configuration properties are available to the hook script as environment variables with the naming pattern `BUILDKITE_PLUGIN_<PLUGIN_NAME>_<CONFIGURATION_PROPERTY>` where `<PLUGIN_NAME>` is not the name defined in `plugin.yml` but a variable-friendly (only letters, numbers and underscores) version of the repository or folder name in uppercase. In this case, the configured value of `pattern` will be available as `BUILDKITE_PLUGIN_FILE_COUNTER_PATTERN`.
```

Reply:
Another alternative would be:

```suggestion
Configuration properties are available to the hook script as environment variables with the naming pattern `BUILDKITE_PLUGIN_<PLUGIN_NAME>_<CONFIGURATION_PROPERTY>` where `<PLUGIN_NAME>` is not the name defined in `plugin.yml` but the repository or folder name compatible with Bash environment variables (in uppercase and only letters, numbers and underscores). In this case, the configured value of `pattern` will be available as `BUILDKITE_PLUGIN_FILE_COUNTER_PATTERN`.
```